.github/workflows/adopt-tapir-ci.yml

```diff
@@ -60,6 +60,20 @@ jobs:
         id: build-docker-image-locally
         run: sbt docker:publishLocal
 
+      - name: Run UI build
+        # run this step only once across all configurations
+        if: matrix.scala == 'Scala3' && matrix.json == 'No'
+        id: run-ui-build
+        run: yarn build
+        working-directory: ./ui
+
+      - name: Run UI tests
+        # run this step only once across all configurations
+        if: matrix.scala == 'Scala3' && matrix.json == 'No'
+        id: run-ui-tests
+        run: yarn test
+        working-directory: ./ui
+
```

Comment:
IMHO it should be either UI build & test here or added to root project in `build.sbt`.
I would prefer the former as we could keep it separated and called when needed IOW it wouldn't impact backend development until one needs it (consider adding that part to `README.md` if it is not there yet) as now it is called two times (both compilation and unit tests).
Here is the [job log](https://github.com/softwaremill/adopt-tapir/runs/7777186105):
1. compilation is part of the `Run unit tests` step
2. UI tests are part of the `Run unit tests` step
3. they are repeated when these two steps are called